### PR TITLE
[Improvement](thrift) optimize thrift messages

### PR DESCRIPTION
**Pull request by Gabriel39**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Now we use a thrift message per fragment instance. However, there are many same messages between instances in a fragment. So this PR aims to extract the same messages and we only need to send thrift message once for a fragment

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.14 seconds
 load time: 517 seconds
 storage size: 17122561215 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230203024319_clickbench_pr_89826.html

File: be/src/exec/data_sink.cpp [hunk 1]
```diff
@@ -177,6 +177,146 @@ Status DataSink::create_data_sink(ObjectPool* pool, const TDataSink& thrift_sink
     return Status::OK();
 }
 
+Status DataSink::create_data_sink(ObjectPool* pool, const TDataSink& thrift_sink,
+                                  const std::vector<TExpr>& output_exprs,
+                                  const TPipelineParams& params, const size_t& local_param_idx,
+                                  const RowDescriptor& row_desc, RuntimeState* state,
+                                  std::unique_ptr<DataSink>* sink, DescriptorTbl& desc_tbl) {
+    DataSink* tmp_sink = nullptr;
+    TPipelineLocalParams local_params = params.local_params[local_param_idx];
```
Comment (HappenLee) [inline be/src/exec/data_sink.cpp:186]: here may be do copy？ use `const auto&`

File: gensrc/thrift/PaloInternalService.thrift [hunk 2]
```diff
@@ -553,3 +553,45 @@ struct TExportStatusResult {
     2: required Types.TExportState state
     3: optional list<string> files
 }
+
```
Comment (HappenLee) [inline gensrc/thrift/PaloInternalService.thrift:556]: Better name
`TPipelineLocalParams` -> `TPipelineInstanceParams`

`TPipelineParams ` -> `TPipelineQueryParams`

Comment (HappenLee) [inline gensrc/thrift/PaloInternalService.thrift:556]: Better name
`TPipelineLocalParams` -> `TPipelineInstanceParams`

`TPipelineParams ` -> `TPipelineQueryParams`

File: be/src/runtime/fragment_mgr.cpp [hunk 3]
```diff
@@ -777,6 +782,157 @@ Status FragmentMgr::exec_plan_fragment(const TExecPlanFragmentParams& params, Fi
...
+                          << " limit: " << PrettyPrinter::print(bytes_limit, TUnit::BYTES);
+            } else {
+                // Already has a query fragments context, use it
+                fragments_ctx = search->second;
+            }
+        }
+    }
+
+    for (size_t i = 0; i < params.local_params.size(); i++) {
+        TPipelineLocalParams local_params = params.local_params[i];
```
Comment (HappenLee) [inline be/src/runtime/fragment_mgr.cpp:878]: same to up

Comment (HappenLee) [inline be/src/runtime/fragment_mgr.cpp:878]: same to up

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**